Python/xYeeshaPages.py

```diff
@@ -158,43 +158,39 @@ def OnGUINotify(self,id,control,event):
-                        YeeshaPageVar.setInt(4)
-                        vault.updatePsnlAgeSDL (psnlSDL)
-
-                        PtSendKIMessageInt(kStartBookAlert,0)
-
-                        if (PageNumber.value) == 25:
-                            #Cleft is done, set SDL to start link back to Relto
-                            actClickableBook.disableActivator()
-                            PtSendKIMessage(kDisableKIandBB,0)
-                            ageSDL = PtGetAgeSDL()
-                            ageSDL["clftIsCleftDone"] = (1,)
-                            vault = ptVault()
-                            vault.addChronicleEntry("CleftSolved",1,"yes")
-                            PtDebugPrint("Chronicle updated with variable 'CleftSolved'.",level=kDebugDumpLevel)
-                            #PtAtTimeCallback(self.key,kWaitFadeoutSecs,kStartFadeoutID)
+            psnlSDL = vault.getPsnlAgeSDL()
+            if psnlSDL:
+                YeeshaPageVar = psnlSDL.findVar("YeeshaPage" + str(PageNumber.value))
+                
+                PtDebugPrint ("xYeeshaPages.py: The previous value of the SDL variable %s is %s" % ("YeeshaPage" + str(PageNumber.value), YeeshaPageVar.getInt()))
 
+                if YeeshaPageVar.getInt() != 0: 
+                    PtDebugPrint ("xYeeshaPages.py: You've already found Yeesha Page #%s. Move along. Move along." % (PageNumber.value))
+                    return
+                    
                 else:
-                    PtDebugPrint("xYeeshaPages: Error trying to access the Chronicle psnlSDL. psnlSDL = %s" % ( psnlSDL))
+                    PtDebugPrint ("xYeeshaPages.py: Yeesha Page #%s is new to you." % (PageNumber.value))
                     
+                    PtDebugPrint ("xYeeshaPages.py: Trying to update the value of the SDL variable %s to 1" % ("YeeshaPage" + str(PageNumber.value)))
+                    YeeshaPageVar.setInt(4)
+                    vault.updatePsnlAgeSDL (psnlSDL)
+
+                    PtSendKIMessageInt(kStartBookAlert,0)
+
+                    if (PageNumber.value) == 25:
+                        #Cleft is done, set SDL to start link back to Relto
+                        actClickableBook.disableActivator()
+                        PtSendKIMessage(kDisableKIandBB,0)
+                        ageSDL = PtGetAgeSDL()
+                        ageSDL["clftIsCleftDone"] = (1,)
+                        vault = ptVault()
+                        vault.addChronicleEntry("CleftSolved",1,"yes")
+                        PtDebugPrint("Chronicle updated with variable 'CleftSolved'.",level=kDebugDumpLevel)
+                        #PtAtTimeCallback(self.key,kWaitFadeoutSecs,kStartFadeoutID)
+
             else:
-                PtDebugPrint("xYeeshaPages: Error trying to access the Vault. Can't access YeeshaPageChanges chronicle." )
+                PtDebugPrint("xYeeshaPages: Error trying to access the Chronicle psnlSDL. psnlSDL = %s" % ( psnlSDL))
```

Comment:
`xYeeshaPages: Error trying to access the Chronicle psnlSDL. psnlSDL = None`?

Reply:
Or at least psnlSDL is falsey. It's like that on master. What do you think should happen here?